src/form.js

```diff
@@ -145,29 +145,49 @@ define([
     }
 
     Form.prototype = {
-        addHashtag: function(hashtag, xpath) {
-            this.hashtagDictionary[hashtag] = xpath;
+        isValidHashtag: function(tag) {
+            return this.hashtagDictionary.hasOwnProperty(this.normalizeHashtag(tag));
+        },
+        addHashtag: function(hashtag, xpath, dontOverwrite) {
+            if (!dontOverwrite || !this.hashtagDictionary[hashtag]) {
+                this.hashtagDictionary[hashtag] = xpath;
+            }
         },
         removeHashtag: function(hashtag) {
             delete this.hashtagDictionary[hashtag];
         },
-        normalizeHashtag: function (xpath_) {
+        transform: function(input, transformFn) {
+            input = this.normalizeEscapedHashtag(input);
+            return escapedHashtags.transform(input, transformFn);
+        },
+        normalize: function (methodName, xpath) {
             // try catch is needed as workaround for having an itemset without
-            // the itemset plugin enabled
+            // the itemset plugin enabled and invalid xpaths
             try {
-                return xpath_ ? this.xpath.parse(xpath_).toHashtag() : xpath_;
+                return xpath ? this.xpath.parse(xpath)[methodName]() : xpath;
             } catch (err) {
-                return xpath_;
+                return xpath.startsWith('#invalid/xpath ') ? xpath.slice(15) : xpath;
```

Comment:
Ah. I see what you mean now. 

Looks like I did this kind of poorly and there are a few edge cases that depend on this. This is here, because it lazily removes #invalid/xpath from any mug properties since they can come from a few places (like parser plugins and copy-paste). I wanted it to be sure that #invalid/xpath was didn't cause an error for the user , but now I'm not sure that's the best option.

It's possible that this should be a code clean up item